src/detect-transform-pcrexform.c

```diff
@@ -61,9 +65,12 @@ static void DetectTransformPcrexformFree(DetectEngineCtx *de_ctx, void *ptr)
 {
     if (ptr != NULL) {
         DetectTransformPcrexformData *pxd = (DetectTransformPcrexformData *) ptr;
+        pcre2_code_free(pxd->regex);
+        pcre2_match_data_free(pxd->match_data);
```

Comment:
I think you need a null check here ... this function can be called with `match_data = NULL`.

Reply:
How could it be `NULL` ?

+ This behaves well for NULL cf https://www.pcre.org/current/doc/html/pcre2_match_data_free.html